slsDetectorSoftware/src/slsDetector.cpp

```diff
@@ -948,124 +940,80 @@ int slsDetector::updateDetector() {
     return ret;
 }
 
-int slsDetector::writeConfigurationFile(const std::string &fname,
-                                        multiSlsDetector *m) {
-    int iline = 0;
-    std::ofstream outfile;
-    outfile.open(fname.c_str(), std::ios_base::out);
-    if (outfile.is_open()) {
-        iline = writeConfigurationFile(outfile, m);
-        outfile.close();
-    } else {
-        throw RuntimeError("Could not open configuration file for writing");
-    }
-    FILE_LOG(logINFO) << iline << " lines written to configuration file";
-    return OK;
-}
+std::vector<std::string> slsDetector::getConfigFileCommands() {
+    std::vector<std::string> base{"hostname",    "port",       "stopport",
+                                    "settingsdir", "fpath",      "lock",
+                                    "zmqport",     "rx_zmqport", "zmqip",
+                                    "rx_zmqip",    "rx_tcpport"};
 
-int slsDetector::writeConfigurationFile(std::ofstream &outfile,
-                                        multiSlsDetector *m) {
-
-    FILE_LOG(logDEBUG1) << "Write configuration file";
-
-    std::vector<std::string> names;
-    // common config
-    names.emplace_back("hostname");
-    names.emplace_back("port");
-    names.emplace_back("stopport");
-    names.emplace_back("settingsdir");
-    names.emplace_back("ffdir");
-    names.emplace_back("outdir");
-    names.emplace_back("lock");
-    // receiver config
-    names.emplace_back("zmqport");
-    names.emplace_back("rx_zmqport");
-    names.emplace_back("zmqip");
-    names.emplace_back("rx_zmqip");
-    names.emplace_back("rx_tcpport");
-
-    // detector specific config
     switch (shm()->myDetectorType) {
     case GOTTHARD:
-        names.emplace_back("detectormac");
-        names.emplace_back("detectorip");
-        names.emplace_back("rx_udpport");
-        names.emplace_back("rx_udpip");
-        names.emplace_back("rx_udpmac");
-        names.emplace_back("rx_hostname");
-
-        names.emplace_back("extsig:0");
-        names.emplace_back("vhighvoltage");
+        base.emplace_back("detectormac");
+        base.emplace_back("detectorip");
+        base.emplace_back("rx_udpport");
+        base.emplace_back("rx_udpip");
+        base.emplace_back("rx_udpmac");
+        base.emplace_back("extsig:0");
```

Comment:
For extsig, we removed the signal index. So, just "extsig" should do.

Reply:
done